strawberry/experimental/pydantic/object_type.py

```diff
@@ -71,11 +71,14 @@ def _build_dataclass_creation_fields(
     else:
         # otherwise we build an appropriate strawberry field that resolves it
         existing_field = existing_fields.get(field.name)
+        graphql_name = None
+        if existing_field and existing_field.graphql_name:
+            graphql_name = existing_field.graphql_name
+        elif field.has_alias and use_pydantic_alias:
```

Comment:
what happens if there `existing_field.graphql_name` and the user passes use_pydantic_alias?

 i think we should error out?  or maybe we can leave it such that existing_field.graphql_name takes precedence (which is how it is now). we'll just need to enforce that behaviour with a test @Matt343 @patrick91

Reply:
maybe erroring out might be best, when would you use graphql_name and use_pydantic_alias together?

Reply:
@patrick91 
```
class User(pydantic.BaseModel):
    age: int
    state: str = Field(alias="_state")
    country: str = Field(alias="_country")

@strawberry.experimental.pydantic.type(User, use_pydantic_alias=True)
class UserType:
    age: strawberry.auto = strawberry.field(name="ageAlias") # graphql is ageAlias
    state: strawberry.auto = strawberry.field(name="state") # graphql is state
    country: strawberry.auto # graphql is _country
```

maybe because use_pydantic_alias is a blanket settings. and the person happens to have a model where there are two pydantic fields with a pydantic alias. but he only wants one field to use the pydantic alias (poor guy)

Reply:
yep! we should write a test to enforce this behaviour too. i can take it in another MR, or @benzolium if he feels like it :D